src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs

```diff
@@ -58,6 +70,110 @@ private async Task<Document> ConvertToEqualsAsync(Document document, TextSpan so
             };
         }
 
+        private async Task<Document> CallOverloadWithEqualityComparerAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var expression = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+            var rawOperation = semanticModel.GetOperation(expression, cancellationToken);
+
+            if (!CompareSymbolsCorrectlyAnalyzer.UseSymbolEqualityComparer(semanticModel.Compilation) ||
+                !semanticModel.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCollectionsGenericIEqualityComparer1, out var iEqualityComparer))
+            {
+                return document;
+            }
+
+            return rawOperation switch
+            {
+                IObjectCreationOperation objectCreation =>
+                    await CallOverloadWithEqualityComparerAsync(
+                        document, objectCreation.Syntax, objectCreation.Constructor, objectCreation.Arguments, isUsedAsExtensionMethod: false,
+                        (generator, args) => generator.ObjectCreationExpression(objectCreation.Type, args), iEqualityComparer, cancellationToken)
+                    .ConfigureAwait(false),
+
+                IInvocationOperation invocation =>
+                    await CallOverloadWithEqualityComparerAsync(
+                        document, invocation.Syntax, invocation.TargetMethod, invocation.Arguments, IsExtensionMethodUsedAsSuch(invocation),
+                        (generator, args) => generator.InvocationExpression(GetExpression(invocation), args), iEqualityComparer, cancellationToken)
+                    .ConfigureAwait(false),
+
+                _ => document,
+            };
+
+            static bool IsExtensionMethodUsedAsSuch(IInvocationOperation invocation)
+                => invocation.IsExtensionMethodAndHasNoInstance()
```

Comment:
I am a little surprise to see no difference on the `Instance` property when the extension method is used as extension or isn't. That's why I am checking that the 1st arg is implicit (which means it is used as extension) or not which means it's used directly.

Reply:
@mavasani Is there a bug in roslyn or is it the intended behavior?

Reply:
@Evangelink This is by design. IOperation is meant to reflect semantics, and the difference you mention is purely syntactic.